File: external_apps/finance-app/app/create-template.js
import * as XLSX from 'xlsx';
import { writeFileSync } from 'fs';

/**
 * Create a template Excel file for DKKB DAN REALISASI
 */
const createTemplateExcel = () => {
    const wb = XLSX.utils.book_new();

    // Create REALISASI sheet with headers
    const realisasiData = [
        ['LAPORAN REALISASI ANGGARAN'],
        [''],
        ['TABEL 1 - SEKSI PENGEMBANGAN'],
        [''],
        ['BAGIAN', ':', 'SEKSI PENGEMBANGAN'],
        ['SUMBER ANGGARAN', ':', 'PAD MURNI'],
        ['PROGRAM', ':', 'PROGRAM PENUNJANG URUSAN PEMERINTAHAN DAERAH PROVINSI'],
        ['Kegiatan', ':', ''],
        ['Sub Kegiatan', ':', ''],
        [''],
        ['NO', 'KODE REKENING', 'URAIAN', 'SEBELUM', 'SETELAH', 'JUMLAH', 'PAPBD', 'JAN', 'FEB', 'MAR', 'APR', 'MEI', 'JUN', 'JUL', 'AGU', 'SEP', 'OKT', 'NOV', 'DES', 'JUMLAH REALISASI'],
        ['', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', ''],
    ];

    const ws_realisasi = XLSX.utils.aoa_to_sheet(realisasiData);

    // Set column widths
    ws_realisasi['!cols'] = [
        { wch: 5 },   // NO
        { wch: 15 },  // KODE REKENING
        { wch: 40 },  // URAIAN
        { wch: 12 },  // SEBELUM
        { wch: 12 },  // SETELAH
        { wch: 12 },  // JUMLAH
        { wch: 12 },  // PAPBD
        { wch: 10 },  // JAN
        { wch: 10 },  // FEB
        { wch: 10 },  // MAR
        { wch: 10 },  // APR
        { wch: 10 },  // MEI
        { wch: 10 },  // JUN
        { wch: 10 },  // JUL
        { wch: 10 },  // AGU
        { wch: 10 },  // SEP
        { wch: 10 },  // OKT
        { wch: 10 },  // NOV
        { wch: 10 },  // DES
        { wch: 12 },  // JUMLAH REALISASI
    ];

    XLSX.utils.book_append_sheet(wb, ws_realisasi, 'REALISASI');

    return wb;
};

// Generate and save the file
const wb = createTemplateExcel();
const buffer = XLSX.write(wb, { type: 'buffer', bookType: 'xlsx' });
writeFileSync('./public/DKKB DAN REALISASI.xlsx', buffer);

console.log('✅ Template Excel file created successfully at: public/DKKB DAN REALISASI.xlsx');

File: external_apps/finance-app/app/create-template.py
#!/usr/bin/env python3
import openpyxl
from openpyxl import Workbook
from openpyxl.styles import Font, Alignment

# Create a new workbook
wb = Workbook()
ws = wb.active
ws.title = "REALISASI"

# Add header rows
data = [
    ['LAPORAN REALISASI ANGGARAN'],
    [''],
    ['TABEL 1 - SEKSI PENGEMBANGAN'],
    [''],
    ['BAGIAN', ':', 'SEKSI PENGEMBANGAN'],
    ['SUMBER ANGGARAN', ':', 'PAD MURNI'],
    ['PROGRAM', ':', 'PROGRAM PENUNJANG URUSAN PEMERINTAHAN DAERAH PROVINSI'],
    ['Kegiatan', ':', ''],
    ['Sub Kegiatan', ':', ''],
    [''],
    ['NO', 'KODE REKENING', 'URAIAN', 'SEBELUM', 'SETELAH', 'JUMLAH', 'PAPBD', 'JAN', 'FEB', 'MAR', 'APR', 'MEI', 'JUN', 'JUL', 'AGU', 'SEP', 'OKT', 'NOV', 'DES', 'JUMLAH REALISASI'],
]

# Write data to worksheet
for row_idx, row_data in enumerate(data, start=1):
    for col_idx, value in enumerate(row_data, start=1):
        cell = ws.cell(row=row_idx, column=col_idx, value=value)
        
        # Bold header row
        if row_idx == 11:
            cell.font = Font(bold=True)
            cell.alignment = Alignment(horizontal='center')

# Set column widths
column_widths = [5, 15, 40, 12, 12, 12, 12, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 12]
for idx, width in enumerate(column_widths, start=1):
    ws.column_dimensions[openpyxl.utils.get_column_letter(idx)].width = width

# Save the workbook
wb.save('public/DKKB DAN REALISASI.xlsx')
print('✅ Template Excel file created successfully at: public/DKKB DAN REALISASI.xlsx')
